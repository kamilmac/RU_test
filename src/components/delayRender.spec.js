import React from 'react'
import { shallow } from 'enzyme'
import DelayRender from './DelayRender'

it('renders DelayRender', () => {
    const props = {
        notes: [
            {
                text: 'test',
            },
        ]
    }
    const wrapper = shallow(<DelayRender props={props} delay={1000} component={<section/>} />)
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('section').length).toEqual(1)
    expect(wrapper.find('section').props().notes).toEqual(props.notes)
})