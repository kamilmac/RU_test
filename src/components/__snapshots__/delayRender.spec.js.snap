exports[`test renders DelayRender 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <section
        notes={
            Array [
                Object {
                  "text": "test",
                },
              ]
        } />
</div>,
  "nodes": Array [
    <div>
      <section
            notes={
                  Array [
                        Object {
                          "text": "test",
                        },
                      ]
            } />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <delayRender
        component={<section />}
        delay={1000}
        props={
                Object {
                        "notes": Array [
                          Object {
                            "text": "test",
                          },
                        ],
                      }
        } />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": delayRender {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "component": <section />,
          "delay": 1000,
          "props": Object {
            "notes": Array [
              Object {
                "text": "test",
              },
            ],
          },
        },
        "refs": Object {},
        "state": Object {
          "props": Object {
            "notes": Array [
              Object {
                "text": "test",
              },
            ],
          },
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <section
                    notes={
                              Array [
                                        Object {
                                          "text": "test",
                                        },
                                      ]
                    } />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <section
                    notes={
                              Array [
                                        Object {
                                          "text": "test",
                                        },
                                      ]
                    } />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <delayRender
    component={<section />}
    delay={1000}
    props={
        Object {
            "notes": Array [
              Object {
                "text": "test",
              },
            ],
          }
    } />,
}
`;
